backend/src/graphql/resolvers/mutations.js

```diff
@@ -87,12 +103,42 @@ module.exports = {
       }
     }
     return User.findOneAndUpdate({ _id: user.id }, args.user, { new: true });
+  },
+  uploadInvoice: async (
+    root,
+    { invoice },
+    { models: { Transaction, Company, Category }, cloudinary }
+  ) => {
+    const { formatData, rules, messages } = uploadInvoiceValidation;
+    await validate(formatData({ ...invoice }), rules, messages);
+    const company = await saveOrRetrieveCompany(invoice.company, Company);
+    invoice.company = company;
+    if (invoice.category && (invoice.category.name || invoice.category.id)) {
+      const category = Category.findOne({
+        $or: [{ _id: company.category.id }, { name: company.category.id }]
+      });
+      if (!category) {
+        throw new Error('Category not found.');
+      }
+      invoice.category = category;
+    }
+    invoice.flow = 'IN';
```

Comment:
Maybe these should be constants declared in a constants file

Reply:
Didn't think about it, yeah it'd better that way.